Lib/fontTools/cffLib/specializer.py

```diff
@@ -267,11 +309,62 @@ def _negateCategory(a):
 	assert a in '0r'
 	return a
 
+def _hasBlendArg(args):
+	for arg in args:
+		if isinstance(arg, list):
+			return True
+	return False
+
+def _convertToBlendCmds(args):
+	# return a list of blend commands, and
+	# the remaining non-blended args, if any.
+	num_args = len(args)
+	stack_use = 0
+	blend_cmds = []
+	blend_args = []
+	i = 0
+	while i < num_args:
+		arg = args[i]
+		if not isinstance(arg, list):
+			blend_args.append(arg)
+			i += 1
+			stack_use += 1
+		else:
+			prev_stack_use = stack_use
+			# The arg is a tuple of blend values.
+			# These are each (master 0,delta 1..delta n)
+			# Combine as many successive tuples as we can,
+			# up to the max stack limit.
+			num_sources = len(arg)
+			blendlist = [arg]
+			i += 1
+			stack_use += 1 + num_sources  # 1 for the num_blends arg
+			while (i < num_args) and isinstance(args[i], list):
+				blendlist.append(args[i])
+				i += 1
+				stack_use += num_sources
+				if stack_use + num_sources > maxStackLimit:
+					# if we are here, max stack is is the CFF2 max stack.
+					break
+			num_blends = len(blendlist)
+			# append the 'num_blends' default font values
+			for arg in blendlist:
+				blend_args.append(arg[0])
+			for arg in blendlist:
+				blend_args.extend(arg[1:])
+			blend_args.append(num_blends)
+			blend_cmds.append(('blend', blend_args))
+			blend_args = []
+			stack_use = prev_stack_use + num_blends
+
+	return blend_cmds, blend_args
```

Comment:
I know spec doesn't allow it, but we should support blends where the args are also blended themselves.  Or at least be able to pass those through without error.